taichi/backends/vulkan/vulkan_device_creator.cpp

```diff
@@ -346,10 +334,33 @@ void VulkanDeviceCreator::pick_physical_device() {
   vkEnumeratePhysicalDevices(instance_, &device_count, devices.data());
   physical_device_ = VK_NULL_HANDLE;
 
-  for (const auto &device : devices) {
-    if (is_device_suitable(device, surface_) && (is_device_visible(device))) {
-      physical_device_ = device;
-      break;
+  for (int i = 0; i < device_count; i++) {
```

Comment:
we have to give the info here, because `vkEnumeratePhysicalDevices` does NOT always print the available devices in the same order. Hence we have to print the info at the point where the device is actually selected.